packages/app-store/googlecalendar/api/webhook.ts

```diff
@@ -0,0 +1,51 @@
+import type { NextApiRequest } from "next";
+
+import { HttpError } from "@calcom/lib/http-error";
+import { defaultHandler, defaultResponder } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+import { credentialForCalendarServiceSelect } from "@calcom/prisma/selects/credential";
+
+import { getCalendar } from "../../_utils/getCalendar";
+
+async function postHandler(req: NextApiRequest) {
+  // 1. validate request
+  if (req.headers["x-goog-channel-token"] !== process.env.CRON_API_KEY) {
+    throw new HttpError({ statusCode: 403, message: "Invalid API key" });
+  }
+
+  /**
+   * TODO: Handle resubscription when channel expires
+    "x-goog-channel-expiration": "Tue, 24 Oct 2023 03:34:14 GMT",
+   */
+  const selectedCalendar = await prisma.selectedCalendar.findFirst({
+    where: {
+      metadata: {
+        path: ["id"],
+        equals: req.headers["x-goog-channel-id"],
+      },
+    },
+    select: {
+      credential: {
+        select: {
+          ...credentialForCalendarServiceSelect,
+          selectedCalendars: {
+            orderBy: {
+              externalId: "asc",
+            },
+          },
+        },
+      },
+    },
+  });
+  if (!selectedCalendar) throw new HttpError({ statusCode: 404, message: "No calendar found" });
+  const { credential } = selectedCalendar;
+  if (!credential) throw new HttpError({ statusCode: 404, message: "No credential found" });
+  const { selectedCalendars } = credential;
```

Comment:
`nitpick`: I don't think all of this de-structuring is accomplishing anything, would be better to just use raw checks and make this more readable?

Eg: `if (selectedCalendar && selectedCalendar?.credential) {...}`

Reply:
I like this approach because it can help debug and points to exactly what's missing. We could add some more details/identifiers in the error message.